engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/CallActivityTest.java

```diff
@@ -279,6 +279,67 @@ public void shouldApplyInputMappings() {
         .contains(tuple("x", "1"), tuple("y", "1"));
   }
 
+  @Test
+  public void shouldOnlyPropagateVariablesDefinedViaInputMappings() {
+    // given
+    ENGINE
+        .deployment()
+        .withXmlResource(
+            "wf-parent.bpmn",
+            parentProcess(
+                c -> c.zeebeInputExpression("x", "y").zeebePropagateAllParentVariables(false)))
+        .deploy();
+
+    // when
+    final var processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(PROCESS_ID_PARENT)
+            .withVariables(Map.of("x", 1))
+            .create();
+
+    // then
+    final var childInstance = getChildInstanceOf(processInstanceKey);
+
+    assertThat(
+            RecordingExporter.variableRecords()
+                .withProcessInstanceKey(childInstance.getProcessInstanceKey())
+                .limit(2))
+        .extracting(Record::getValue)
+        .extracting(v -> tuple(v.getName(), v.getValue()))
+        .contains(tuple("y", "1"))
+        .doesNotContain(tuple("x", "1"));
```

Comment:
```suggestion
        .containsExactly(tuple("y", "1"));
```

🙃 Slightly, nitpicky but we can combine this into one. `containsExactly` checks if the result only contains the given element and in order.